compact_bytes/src/fixed.rs

```diff
@@ -0,0 +1,471 @@
+        let mut pointer = self as *const Self as *const u8;
+        if self.spilled() {
+            pointer = unsafe { self.heap.ptr }.as_ptr()
+        }
+        pointer
+    }
+
+    #[inline(always)]
+    fn as_mut_ptr(&mut self) -> *mut u8 {
+        // SAFETY: `InlineBytes15` and `HeapBytesFixed` have the same size and alignment.
+        //
+        // Note: This code is very carefully written so we can benefit from branchless
+        // instructions.
+        let mut pointer = self as *mut Self as *mut u8;
+        if self.spilled() {
+            pointer = unsafe { self.heap.ptr }.as_ptr()
+        }
+        pointer
+    }
+}
+
+impl Default for CompactBytesSlice {
+    #[inline]
+    fn default() -> Self {
+        CompactBytesSlice::new(&[])
+    }
+}
+
+impl Deref for CompactBytesSlice {
+    type Target = [u8];
+
+    #[inline]
+    fn deref(&self) -> &[u8] {
+        self.as_slice()
+    }
+}
+
+impl DerefMut for CompactBytesSlice {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut [u8] {
+        self.as_mut_slice()
+    }
+}
+
+impl AsRef<[u8]> for CompactBytesSlice {
+    #[inline]
+    fn as_ref(&self) -> &[u8] {
+        self.as_slice()
+    }
+}
+
+impl<T: AsRef<[u8]>> PartialEq<T> for CompactBytesSlice {
+    #[inline]
+    fn eq(&self, other: &T) -> bool {
+        self.as_slice() == other.as_ref()
+    }
+}
+
+impl Eq for CompactBytesSlice {}
+
+impl Hash for CompactBytesSlice {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.as_slice().hash(state)
+    }
+}
+
+impl fmt::Debug for CompactBytesSlice {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:?}", self.as_slice())
+    }
+}
+
+impl Drop for CompactBytesSlice {
+    #[inline]
+    fn drop(&mut self) {
+        // Note: we hint to the compiler that dropping a heap variant is cold to improve the
+        // performance of dropping the inline variant.
+        #[cold]
+        fn outlined_drop(this: &mut CompactBytesSlice) {
+            let heap = unsafe { &mut this.heap };
+            heap.dealloc();
+        }
+
+        if self.spilled() {
+            outlined_drop(self);
+        }
+    }
+}
+
+impl Clone for CompactBytesSlice {
+    #[inline]
+    fn clone(&self) -> Self {
+        // Note: we hint to the compiler that cloing a heap variant is cold to improve the
+        // performance of cloning the inline variant.
+        #[cold]
+        fn outlined_clone(this: &CompactBytesSlice) -> CompactBytesSlice {
+            CompactBytesSlice::new(this.as_slice())
+        }
+
+        if self.spilled() {
+            outlined_clone(self)
+        } else {
+            let inline = unsafe { &self.inline };
+            CompactBytesSlice { inline: *inline }
+        }
+    }
+}
+
+impl From<Box<[u8]>> for CompactBytesSlice {
+    #[inline]
+    fn from(value: Box<[u8]>) -> Self {
+        if value.is_empty() {
+            let inline = InlineBytes15::empty();
+            return CompactBytesSlice { inline };
```

Comment:
You could return `CompactBytesSlice::empty()` here.

Reply:
Updated!